lib/web/apiserver.go

```diff
@@ -3731,12 +3752,102 @@ func (h *Handler) WithClusterAuth(fn ClusterHandler) httprouter.Handle {
 	})
 }
 
+func (h *Handler) writeErrToWS(ws *websocket.Conn, err error) {
+	errEnvelope := Envelope{
+		Type:    defaults.WebsocketError,
+		Payload: err.Error(),
```

Comment:
We should try an extract `trace.UserMessage` first. Sending an arbitrary error all the way to the frontend is likely to result in poor UX or even divulging info that shouldn't be.